### PR TITLE
(feat) Improve EditTeam flow 

### PR DESCRIPTION
**Pull request by brrkrmn**
## Summary

The EditTeam view's code involved unused functions and variables, and the design was not consistent with the EditProfile view. Additionally, I implemented the code to show error message when the user tries to submit the form without a  team name. 

## Changes for the appearance

- Edit Team button label is changed from "Edit Team Info" to "Edit Team"
- The input headings were removed since the labels convey the same text
- A description text is added for the bio field
- Validation is added so that empty username field shows an error
- Changed the submit button's label from "Edit Profile" to "Save Changes"
- Add a button to discard changes
- Edited delete team button style and removed the description since it is shown inside the modal

PS: I used hardcoded texts for some labels and Typography elements. This is because in my other PR #843, I handled all the hardcoded texts to be used from the translation files and created new keys for this file. I could not use those keys because the PR may not be merged. I also did not want to do the same steps twice not waste time. I will change the texts of EditTeam.jsx and EditTeamScripts.js as soon as that #843 is resolved.

## Changes regarding code refactoring

I noticed that the code in EditTeam.jsx and EditTeamScripts.js is copied from the EditProfile view and has many parts that should be changed or removed. These are the changes to simplify and refactor this code:

- EditProfile form's refs were deleted and two new refs for EditTeam form is created
- Unused functions and variables were removed
- In the old code, form values were saved in the state manually. Now, Formik's `handleChange` prop is used. Thus, the old state is removed.
- The handleSubmit function is moved to EditTeamScripts.js EditProfile function so that the form submits after all the validations run.
- New id's and names were given to the form elements because the existing id's and refs caused the bio field to get the data from the user's profile. 
- To prevent previous complications to happen, `getProfile()` function in the effect is removed because the code does not use any of its return values.
- Edited getTeamProfile function to fill the username and bio fields if they have an initial value
- Validations are added for the form elements

## The backend change

After using the Formik props, the form did not submit when the user left the bio field empty. In the frontend, the bio field is sent to the backend as empty string if the user sent it empty. The form worked as expected whwn the bio is filled. When I checked edit profile form's bio field, it is also sent as an empty string when submitted empty. Thus, I could not prevent this from happening with changes to the frontend. 

https://stackoverflow.com/a/55150513 => Assuming that the problem is caused because Django interprets empty string as falsy, I changed the line to raise a validation error only if the groupname is missing.

I am not really sure about this solution, looking forward to reviews 🙂

## Videos

Before

https://github.com/unstructuredstudio/zubhub/assets/92817363/2405c0b7-c382-4a49-b722-989052be7c4a

After

https://github.com/unstructuredstudio/zubhub/assets/92817363/209e3693-3a21-4207-af22-69a7ab9e2daa

### CODE REVIEW

**Comment by brrkrmn** [comment]
For future improvement, some functions' names can be changed to be more appropriate for the view. Instead of editProfile => editTeam etc.

File: zubhub_frontend/zubhub/src/views/edit_team/EditTeam.jsx [hunk 1]
```diff
@@ -308,47 +257,63 @@ function EditTeam(props) {
...
+                  <Grid item xs={12}>
+                    <Link to={`/teams/${groupname}`} className={classes.textDecorationNone}>
+                      <CustomButton
+                        variant="outlined"
+                        size="large"
+                        secondaryButtonStyle
+                        customButtonStyle
+                        fullWidth
+                      >
+                        {'Discard Changes'}
```
Comment (coderatomy) [inline zubhub_frontend/zubhub/src/views/edit_team/EditTeam.jsx]: Consider translating this

File: zubhub_frontend/zubhub/src/views/edit_team/EditTeam.jsx [hunk 2]
```diff
@@ -308,47 +257,63 @@ function EditTeam(props) {
...
+                    variant="outlined"
+                    size="large"
+                    secondaryButtonStyle
+                    dangerButtonStyle
+                    fullWidth
+                    onClick={() =>
+                      handleSetState(handleToggleDeleteAccountModal(state))
+                    }
+                  >
+                    {'Delete Team'}
```
Comment (coderatomy) [inline zubhub_frontend/zubhub/src/views/edit_team/EditTeam.jsx]: same here

**Review by coderatomy — CHANGES_REQUESTED** [review]
Nice work @brrkrmn. Just a few thing you missed out
Also consider resolving the conflicts

File: zubhub_frontend/zubhub/src/views/edit_team/EditTeam.jsx (hunk 1 above)
Comment (brrkrmn) [reply zubhub_frontend/zubhub/src/views/edit_team/EditTeam.jsx]: Actually I was waiting for another PR to be merged to translate the texts. That is merged, so I am working on translating these as well, thank you! 🙂

**Comment by brrkrmn** [comment]
@shrish, I replaced the texts with translations. This is ready for review ☺️

**Comment by tuxology** [comment]
@brrkrmn Please refer the Figma design to see edit team implementation:

image
 

You can use existing styles components from what is being done in #778 I agree to your other points about code refactoring, but lets take correct components from similar styled pages such as Create Team or from work done in #778. In addition, ensure this works in responsive mode as shown in screenshots

**Comment by brrkrmn** [comment]
@tuxology Hey, I am currently working on this according to the Figma design. I see that editTeam.jsx's styles are imported from efitProfileStyles.js. Should I create a new editTeamStyles.js file to separate the view's design?

**Comment by brrkrmn** [comment]
@tuxology, I've changed the files according to the Figma design. 
I created an editTeamStyles.js file, which is the duplicate of the existing styles except for the unused parts. I also added new classes for the dialog part. 
In the Figma design, there is not a submit button but the error handling and validation in this form requires one, so I kept the submit button. 
Here is the final version:
Screenshot 2023-10-16 at 17 48 04
Screenshot 2023-10-16 at 17 48 25

And the mobile version:
Screenshot 2023-10-16 at 17 47 33
Screenshot 2023-10-16 at 17 47 40

**Review by tuxology — APPROVED** [review]
LGTM 🌈

**Review by tuxology — CHANGES_REQUESTED** [review]
Upon closer look, have a few comments: 

Is it possible to use https://github.com/unstructuredstudio/zubhub/blob/master/zubhub_frontend/zubhub/src/components/modals/Modal.jsx#L16 for your modal instead of creating new styles? If you would like to change the design, its fine, lets change it in a single place. I am re-marking the PR for changes.

**Comment by brrkrmn** [comment]
@tuxology I used our Modal component for the dialog. I moved the dialog styles to the component's style file but did not style the component itself to avoid changing other modals in the app. This view now uses classes from the modal component, but for reusability, I think some changes can be made to the modal component in terms of styling.

**Review by tuxology — APPROVED** [review]

**Pull request merged**